cmake.md

```diff
@@ -0,0 +1,30 @@
+## CMake build system
+### Terminology
+
+The CMake system in symbiflow-arch-defs uses the following hierarchy:
+
+* ARCH - The architecture of FPGA.  This is used to capture architecture level parameters (e.g. tech libraries).
+* DEVICE_TYPE - A type of device.  This is defined as an ARCH and the arch.xml which defines the units with this device type.
+* DEVICE - A specific instance of a DEVICE_TYPE.  A device definition is a DEVICE_TYPE, a PACKAGE list, and IO map definitions for each package if needed.
+* BOARD - A specific DEVICE and PACKAGE instance, along with a PROG_TOOL/PROG_CMD.  A board is what is targeted by ADD_FPGA_TARGET.
```

Comment:
The last sentence is a bit strange to me? Maybe I'm just confused by the name `ADD_FPGA_TARGET`.

Reply:
ADD_FPGA_TARGET is our version of CMake's ADD_EXECUTABLE.  An ADD_FPGA_TARGET targets a board (e.g. build these sources for this board).

Reply:
Updated doc.